.github/workflows/partial-tests.yml

```diff
@@ -0,0 +1,112 @@
+name: Partial Tests Workflow
+
+on:
+  push:
+    branches: [ develop, dev/*, release/* ]
+  pull_request:
+    branches: [ develop, dev/*, release/* ]
+
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    steps:
+      # replace for Ruff in the future
+      - uses: actions/checkout@v4
+      - uses: ricardochaves/python-lint@v1.4.0
+        with:
+          use-pylint: false
+          use-isort: false
+          use-mypy: false
+          extra-black-options: "--line-length=120 --diff"
+          extra-pycodestyle-options: "--max-line-length=120  --exclude=tests/gui --ignore=E121,E123,E126,E226,E24,E704,W503,W504,E203"
+          extra-mypy-options: "--ignore-missing-imports --implicit-optional --no-namespace-packages --exclude (taipy/templates/|generate_pyi.py|tools) --follow-imports skip"
+          extra-isort-options: "--line-length=120 --force-grid-wrap=10 --multi-line=VERTICAL_HANGING_INDENT --trailing-comma"
```

Comment:
Just by curiosity why did you moved linter from a separate workflow to the partial-tests workflow?

Reply:
Linter is usually one of the first steps on a CI pipeline. I think it makes more sense to build it like a workflow together with unit tests and first execute the linter, if that step fails, unit tests aren't executed. For me it makes more manageable grouping related jobs into a single workflow instead of managing separate files.